[not for merge - just surface issue][inductor] Fix bug in group batch fusion pass

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #124501
* __->__ #125936

@jackiexu1992 Wondering if you could fix the test case after this change. I think pattern matcher should still work.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

anijain2305 committed:
[inductor] Fix bug in group batch fusion pass

[ghstack-poisoned]

mengluy0125 commented on test/inductor/test_group_batch_fusion.py:
```diff
@@ -382,7 +382,10 @@ def test_pointwise_op_fusion(self):
         input = [torch.randn(50, 1000, requires_grad=True, device="cuda")]
-        traced = torch.compile(module)
+
+        def wrapper(*args, **kwargs):
```
Please add the following specific configs to the test func.

    @requires_cuda
    @torch._inductor.config.patch(
        pre_grad_fusion_options={
            "batch_tanh": {},
            "batch_relu": {},
            "batch_sigmoid": {},
        },
        post_grad_fusion_options={
            "batch_aten_add": {},
            "batch_aten_mul": {},
            "batch_aten_sub": {},
            "batch_aten_div": {},
        },
    )
def test_pointwise_op_fusion(self):
....

mengluy0125 replied:
@requires_cuda
  @torch._inductor.config.patch(
      pre_grad_fusion_options={
          "batch_relu": {},
          "batch_sigmoid": {},
          "batch_tanh": {},
      },
      post_grad_fusion_options={
          "batch_aten_div": {},
          "batch_aten_sub": {},
          "batch_aten_mul": {},
          "batch_aten_add": {},
      },
  )
  def test_pointwise_op_fusion(self):
      counters.clear()
      module = TestPoitwiseOps("cuda")
      input = [torch.randn(50, 1000, requires_grad=True, device="cuda")]
      def wrapper(*args, **kwargs):
          return module(*args, **kwargs)
      traced = torch.compile(wrapper)
      ref = module(*input)
      res = traced(*input)
      self.compare_pred(module, traced, input)
      self.assertEqual(counters["inductor"]["batch_tanh"], 1)
      self.assertEqual(counters["inductor"]["batch_relu"], 1)
      self.assertEqual(counters["inductor"]["batch_sigmoid"], 1)
      self.assertEqual(counters["inductor"]["batch_aten_add"], 1)
      self.assertEqual(counters["inductor"]["batch_aten_mul"], 1)
      self.assertEqual(counters["inductor"]["batch_aten_sub"], 1)
      self.assertEqual(counters["inductor"]["batch_aten_div"], 1)
      self.assertTrue(torch.allclose(ref, res, equal_nan=True))
      counters.clear()


The test was passed in my local run

Test UI: https://www.internalfb.com/intern/testinfra/testrun/9007199288699223
Network: Up: 52KiB  Down: 11KiB  (reSessionID-318dbc7c-1b10-448e-9ab5-28560697c0fa)
Jobs completed: 8. Time elapsed: 1:40.3s.
Cache hits: 0%. Commands: 2 (cached: 0, remote: 0, local: 2)
Tests finished: Pass 1. Fail 0. Fatal 0. Skip 0. Build failure 0

anijain2305 committed:
Update on "[not for merge - just surface issue][inductor] Fix bug in group batch fusion pass"


jackiexu1992 Wondering if you could fix the test case after this change. I think pattern matcher should still work.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

anijain2305 committed:
Update on "[not for merge - just surface issue][inductor] Fix bug in group batch fusion pass"


jackiexu1992 Wondering if you could fix the test case after this change. I think pattern matcher should still work.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

anijain2305 committed:
Update on "[not for merge - just surface issue][inductor] Fix bug in group batch fusion pass"


jackiexu1992 Wondering if you could fix the test case after this change. I think pattern matcher should still work.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

anijain2305 commented:
Closing this - changes merged in https://github.com/pytorch/pytorch/pull/124501
 Thanks @mengluy0125 for all the help.
